clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -568,7 +568,7 @@
     final Metrics metrics;
     final KafkaConsumerMetrics kafkaConsumerMetrics;
 
-    private final Logger log;
+    private Logger log;
```

Comment:
Why did we make this not final?

Reply:
Because when an error is thrown at initializing groupInstanceConfig, the log would not be initialized yet where we checked it ` == null` below.